pkg/kudoctl/cmd/diagnostics/print.go

```diff
@@ -0,0 +1,195 @@
+package diagnostics
+
+import (
+	"fmt"
+	"io"
+	"path/filepath"
+	"strings"
+
+	"github.com/kudobuilder/kudo/pkg/kudoctl/util/kudo"
+
+	"github.com/spf13/afero"
+	"gopkg.in/yaml.v2"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/cli-runtime/pkg/printers"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	DiagDir = "diag"
+	KudoDir = "diag/kudo"
+)
+
+type printMode int
+
+const (
+	ObjectWithDir      printMode = iota // print object into its own nested directory based on its name and kind
+	ObjectListWithDirs                  // print each object into its own nested directory based on its name and kind
+	RuntimeObject                       // print as a file based on its kind only
+)
+
+// nonFailingPrinter - print provided data into provided directory and accumulate errors instead of returning them.
+// Creates a nested directory if an object type requires so.
```

Comment:
```suggestion
// Creates parent directories if they are required and do not exist.
```
Is my suggestion correct?

Reply:
I agree that my description is not clear, however the alternative may not be clear as well. So, unwrapping the description:
The main idea is, that in some cases we'd like to wrap an object into its own directory which I called **"nested"**, e.g. we'd like `pod_zookeeper-instance-zookeeper-1/zookeeper-instance-zookeeper-1.yaml`, but on the contrast: simply `servicelist.yaml`. Printer determines this via the `printMode` parameter. If the provided _parent directory_ (in this case e.g `diag/operator_zookeeper/instance_zookeeper-instance`) has not yet been created for some reason, then MkdirAll takes care of it, but that's just another thing which seemed to me less important and kind of self-assuming.